sysroot_creator/scripts/create_cc_sysroot.py

```diff
@@ -142,184 +147,160 @@ def __str__(self):
+        image_tag = self._platform.get_workspace_image_tag()
+        logger.info('Exporting filesystem of image {} into tarball {}'.format(
+            image_tag, sysroot_tarball_path))
+
+        try:
+            sysroot_container = DOCKER_CLIENT.containers.run(image=image_tag, detach=True)
+            with open(str(sysroot_tarball_path), 'wb') as out_f:
+                out_f.writelines(sysroot_container.export())
+            sysroot_container.stop()
+            with tarfile.open(str(sysroot_tarball_path)) as sysroot_tar:
+                relevant_dirs = ['lib', 'usr', 'etc', 'opt', 'root_path', 'ros2_ws/install']
+                relevant_members = (
+                    m for m in sysroot_tar.getmembers()
+                    if re.match('^({}).*'.format('|'.join(relevant_dirs)), m.name) is not None
+                )
+                sysroot_tar.extractall(str(self._target_sysroot), members=relevant_members)
+        finally:
+            shutil.rmtree(tmp_sysroot_dir, ignore_errors=True)
+
+        logger.info('Success exporting sysroot to path [{}]'.format(self._target_sysroot))
+
+    def _write_cc_build_setup_script(self) -> Path:
+        """Create setup file for cross-compile build."""
+        cc_build_setup_file_path = self._target_sysroot / 'cc_build_setup.bash'
+        cc_build_setup_file_contents = CC_BUILD_SETUP_FILE_TEMPLATE.render(
+            target_arch=self._platform.arch,
+            target_triple=self._platform.cc_toolchain,
+            cc_root=self._target_sysroot,
+            ros_root='{cc_root_dir}/sysroot/opt/ros/{distro}'.format(
+                cc_root_dir=self._target_sysroot, distro=self._platform.distro))
+        with open(str(cc_build_setup_file_path), 'w') as out_f:
+            out_f.write(cc_build_setup_file_contents)
+        return cc_build_setup_file_path
+
+    def _write_cc_system_setup_script(self) -> Path:
+        """Create setup file for sysroot setup."""
+        cc_system_setup_script_path = self._target_sysroot / 'cc_system_setup.bash'
+        cc_system_setup_script_contents = CC_BUILD_SYSTEM_SETUP_SCRIPT_TEMPLATE.render(
+            cc_root=self._target_sysroot,
+            target_triple=self._platform.cc_toolchain)
+        with open(str(cc_system_setup_script_path), 'w') as out_f:
+            out_f.write(cc_system_setup_script_contents)
+        return cc_system_setup_script_path
+
+    def write_setup_scripts(self) -> None:
+        """Write both the build and system setup scripts."""
+        self.system_setup_script_path = self._write_cc_build_setup_script()
+        self.build_setup_script_path = self._write_cc_system_setup_script()
+
+    def setup_sysroot_environment(self) -> None:
+        """Set up the environment with variables and symbolic links."""
+        logger.info('Sourcing sysroot environment...')
+        logger.info("Executing 'bash {}'".format(self.system_setup_script_path))
+        subprocess.run(['bash', str(self.system_setup_script_path)])
```

Comment:
You don't need bash if this script as a correct shebang.

Reply:
Will update in another PR that changes the script templates to use the built in Python `Template` library.